hypervisor/arch/x86/sgx.c

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (C) 2019 Intel Corporation. All rights reserved.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include <types.h>
+#include <errno.h>
+#include <cpufeatures.h>
+#include <cpu_caps.h>
+#include <sgx.h>
+#include <cpuid.h>
+#include <vm.h>
+#include <logmsg.h>
+
+#define SGX_OPTED_IN (MSR_IA32_FEATURE_CONTROL_SGX_GE | MSR_IA32_FEATURE_CONTROL_LOCK)
+
+/* For the static variables, which are not explicitly initialzed will be inited to 0 */
+static int32_t init_sgx_ret = 0;
+static struct epc_section pepc_sections[MAX_EPC_SECTIONS];			/* physcial epc sections */
+static struct epc_map vm_epc_maps[MAX_EPC_SECTIONS][CONFIG_MAX_VM_NUM];		/* epc resource mapping for VMs */
+
+/* Partition physcial EPC resource according to VM configurations.
+ * Build the mappings between HPA and GPA for EPT mapping later.
+ * EPC resource partition and mapping relationship will stay the same after sgx init.
```

Comment:
“stay the same”  --> stay unchanged or will be not changed.

Reply:
Will update it.